### ISSUE TITLE
Navigation bar shows sometimes not the entries

### ISSUE THREAD

**Issue by hpetriffer** [X-Sharp/XSharpPublic#1041]
When you create a new class and add a method, the Navigation Bar does not jump to that method.
![image](https://user-images.githubusercontent.com/20905638/170438093-4f7ce629-0294-4eea-a02d-14848bf1d2c5.png)
If you select the displayed method, then "Global scope" is displayed
![image](https://user-images.githubusercontent.com/20905638/170438394-a389ec18-b5e6-4346-8fe0-f8fa20aecf5d.png)
When you close the file and open it again it works.

Another problem I have sometimes observed is that in partial classes, where the case does not match, the selected entry and the jump usually do not match.

**Comment by RobertvanderHulst** [comment]
This seems to be a timing issue. 
The editor responds to a message that indicates that the contents of the buffer was changed and has been parsed into entities. This message was sent by the parser just before the contents was saved in the database.
Because your class is PARTIAL and you have enable the option to include also the members of the same class in other files, the editor tries to populate the combobox from the Intellisense database. 
If you have a non partial class then the contents from the combobox is directly retrieved from the parser.
I made a change in the parser to send the "ContentsChanged" message after the entities are saved to the database
